jetstream/config.py

```diff
@@ -627,6 +627,20 @@ def resolve(self, spec: "AnalysisSpec", experiment: ExperimentConfiguration) ->
 MetricsConfigurationType = Dict[AnalysisPeriod, List[Summary]]
 
 
+def validate_metric_definitions(instance: Any, attribute: Any, value: Any) -> None:
+    known_keys = instance.__dict__.keys()
+    for metric_definition in instance.definitions.keys():
+        for k in known_keys:
+            if k != "definitions":
+                if (
+                    metric_definition in [m.name for m in getattr(instance, k)]
+                    or metric_definition == "active_hours"
```

Comment:
@scholtzan Not quite sure what is going on here, maybe this is related to `active_hours` being a default metric? With this validation I observed the following test failures:
```
FAILED jetstream/tests/test_external_config.py::TestExternalConfig::test_from_github_repo - Exception: Metric definition: active_hours never used
FAILED jetstream/tests/test_metadata.py::test_metadata_from_config - Exception: Metric definition: active_hours never used
FAILED jetstream/tests/test_metadata.py::test_metadata_reference_branch - Exception: Metric definition: active_hours never used
FAILED jetstream/tests/test_metadata.py::test_metadata_from_config_missing_metadata - Exception: Metric definition: active_hours never used
FAILED jetstream/tests/test_metadata.py::test_export_metadata - Exception: Metric definition: active_hours never used
```

Which also is interesting that it's only for those few tests/configs and not all of them. For now I "fixed" it be explicitly ignoring `active_hours` in validation. Do you have any ideas what is going on here?

Reply:
Mh I see what's happening here. We do have a default config defined in https://github.com/mozilla/jetstream/blob/main/jetstream/config/default_metrics.toml which we [merge with the custom config](https://github.com/mozilla/jetstream/blob/4ca2d3e217dde6722ccc63cf40255c71aa5ded32/jetstream/cli.py#L232) when we run the experiment.
However, for getting the metadata we don't merge the external config with the default config: https://github.com/mozilla/jetstream/blob/4ca2d3e217dde6722ccc63cf40255c71aa5ded32/jetstream/external_config.py#L115-L118
So if there is a custom config that overwrites one of the default metrics, then this error happens.

I'm fine with dropping the check of all metric definitions being used since otherwise to make this work it will get quite hairy. The most important thing to implement is that we are case-insensitive which the rest of the PR seems to be doing.